conductr_cli/sandbox_run_jvm.py

```diff
@@ -148,106 +153,114 @@ def obtain_sandbox_image(image_dir, image_version):
     :param image_version: the version of the sandbox to be downloaded.
     :return: the pair containing path to the expanded core directory and path to the expanded agent directory
     """
-    def resolve_core_binary_from_cache():
+    def resolve_binary_from_cache(binary_cache_path):
         """
-        Checks for the presence of `${cache_dir}/core/${image_version}.tar.gz`.
+        Checks for the presence of the ConductR binary in the cache directory.
 
+        :param binary_cache_path the path of the cached ConductR universal binary.
         :return: If present, return the path to the binary file, else return None.
         """
-        pass
-
-    def resolve_agent_binary_from_cache():
-        """
-        Checks for the presence of `${cache_dir}/agent/${image_version}.tar.gz`.
-
-        :return: If present, return the path to the binary file, else return None.
-        """
-        pass
+        if os.path.exists(binary_cache_path):
+            return binary_cache_path
+        else:
+            return None
 
-    def download_core_from_bintray(bintray_auth):
+    def download_from_bintray(bintray_auth, bintray_package_name):
         """
-        Downloads core from bintray given `${image_version}`.
+        Downloads ConductR binary from Bintray given `${binary_info}`.
 
-        The images are expected to be cached in `${cache_dir}/core`.
+        The images are expected to be cached in `${image_dir}`.
 
-        If the binary is not yet available within the local cache, then it will be downloaded from Bintray.
-
-        The aftefacts will be available under the following Bintray repo:
+        The artifacts are available under the following Bintray repo:
 
         https://bintray.com/lightbend/commercial-releases/ConductR-Universal
 
         As part of the download:
         - A progress bar will be displayed.
-        - The download will be saved into `${cache_dir}/core/${image_version}.tar.gz.tmp`. Once download is complete,
-          this file will be moved to `${cache_dir}/core/${image_version}.tar.gz`.
+        - The download will be saved into `${image_dir}/${filename}.tgz.tmp`.
+        Once download is complete, this file will be moved to `${image_dir}/${filename}.tgz`.
 
         :param bintray_auth: a tuple containing Bintray auth information which can be passed into the request library.
                              This information is loaded from bintray credentials file, which can be None if the
                              credentials is not specified
+        :param bintray_package_name: the Bintray package name of the ConductR binary
         :return: path to the downloaded file
         """
-        pass
+        _, _, cached_file = bintray_resolver.bintray_download(image_dir, BINTRAY_LIGHTBEND_ORG, BINTRAY_CONDUCTR_REPO,
```

Comment:
This error case is now handled. Additionally, other error cases are now handled too. For more information check out https://github.com/typesafehub/conductr-cli/pull/238#issue-199929586.